src/Xamarin.Android.Build.Tasks/Tasks/Aot.cs

```diff
@@ -239,11 +239,11 @@ int GetNdkApiLevel(string androidNdkPath, string androidApiLevel, AndroidTargetA
 
 			var task = ThreadingTasks.Task.Run ( () => {
 				return RunParallelAotCompiler (nativeLibs);
-			});
+			}, Token);
 
 			task.ContinueWith ( (t) => {
 				Complete ();
-			});
+			}).ConfigureAwait (false);
```

Comment:
I'm not sure `ConfigureAwait(false)` does anything unless you are awaiting it. I think it only controls the code *after* an `await` statement.

I think you need to passing the `options` to an overload of `ContinueWith`: https://msdn.microsoft.com/en-us/library/dd321561(v=vs.110).aspx

Probably not a way to write a unit test for this to know if it's for sure correct...